src/verifiers/verificationBuilder.ts

```diff
@@ -1,21 +1,31 @@
-import { VerificationFragment, VerificationManagerOptions, Verifier } from "../types/core";
+import {
+  VerificationBuilderOptions,
+  VerificationFragment,
+  PromiseCallback,
+  Verifier,
+  VerifierOptions,
+} from "../types/core";
+import { getProvider } from "../common/utils";
 
 /**
  * A verification manager will run a list of {@link Verifier} over a signed document.
  * Before running each verifier, the manager will make sure the verifier can handle the specific document by calling its exposed test function.
  * The manager will return the consolidated list of {@link VerificationFragment}
  */
-export const verificationBuilder = <Document>(verifiers: Verifier<Document>[]) => (
-  document: Document,
-  options: VerificationManagerOptions
-): Promise<VerificationFragment[]> => {
+export const verificationBuilder = <Document>(
+  verifiers: Verifier<Document>[],
+  builderOptions: VerificationBuilderOptions
```

Comment:
This is a breaking change.

Now that I think about it even the change of signature of verifier is a breaking change (ie for the one using a custom verifier this won't work anymore)

Reply:
Yes.. thats why all the test files are updated. The design considerations behind this change is that when we define or build the verify function we know which network it is on already, so it shouldnt be the onus of the `verify` caller to define the network again, this way you wont have the `network` strewn all over the caller code.